app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -103,8 +103,12 @@ class BrowserActivity : DuckDuckGoActivity(), CoroutineScope {
         viewModel.awaitClearDataFinishedNotification()
     }
 
-    override fun onDestroy() {
+    override fun onStop() {
```

Comment:
Double-checking this, I saw that we are executing fragment transactions inside our coroutine, and so, we need to cancel any work at this point.
When activity goes to background, `onDestroy` won't be called (if activity is not killed), only `onStop`.